waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim

```diff
@@ -572,6 +543,44 @@ when defined(rlnzerokit):
     var rootValue = cast[ptr MerkleNode] (root.`ptr`)[]
     return ok(rootValue)
 
+
+proc updateValidRootQueue*(wakuRlnRelay: WakuRLNRelay, root: MerkleNode): void =
+  # updates the valid merkle root queue with the latest root, and pops the oldest one
+  let overflowCount = wakuRlnRelay.validMerkleRoots.len() - AcceptableRootWindowSize
+  if overflowCount >= 0:
+    # Delete the oldest n elements in the deque (index 0..n)
+    for i in 0..overflowCount:
+      wakuRlnRelay.validMerkleRoots.popFirst() 
+  # Push next root into the queue
+  wakuRlnRelay.validMerkleRoots.addLast(root)
+
+proc insertMember*(wakuRlnRelay: WakuRLNRelay, idComm: IDCommitment): RlnRelayResult[void] =
+  # inserts a new id commitment into the local merkle tree, and adds the changed root to the 
+  # queue of valid roots
+  let actionSucceeded = wakuRlnRelay.rlnInstance.insertMember(idComm)
+  if not actionSucceeded:
+    return err("could not insert id commitment into the merkle tree")
+
+  let rootAfterUpdate = ?wakuRlnRelay.rlnInstance.getMerkleRoot()
+  wakuRlnRelay.updateValidRootQueue(rootAfterUpdate)
+  return ok()
+  
+
+proc removeMember*(wakuRlnRelay: WakuRLNRelay, index: MembershipIndex): RlnRelayResult[void] =
+  # removes a commitment from the local merkle tree at `index`, and adds the changed root to the
+  # queue of valid roots
```

Comment:
```suggestion
  ## removes a commitment from the local merkle tree at `index`, and adds the changed root to the
  ## queue of valid roots
```

Reply:
Addressed in [490908d](https://github.com/status-im/nwaku/pull/1177/commits/490908da8efbf14b14d767c4e9d874fad1f94264)